### ISSUE TITLE
in regard to the use of NWJS

### ISSUE THREAD

**Issue by LiaoSept** [facebook/draft-js#194]
![](https://cloud.githubusercontent.com/assets/15375011/13695704/35554b50-e79c-11e5-9616-ae9944485352.png)

due to `global` NWJS under the `Node Context`

-------------------------
HACK: [Monkey-patch](https://en.wikipedia.org/wiki/Monkey_patch)
```javascript
(function () {
    global.getSelection = window.getSelection.bind(window);
}());
```
-------------------------
see http://nwjs.io/
see https://github.com/facebook/draft-js/search?utf8=%E2%9C%93&q=global.getSelection&type=Code

**Comment by zpao** [comment]
Are you using the node module or the dist file? If you're using the node module, how are you packaging it?

**Comment by spicyj** [comment]
I would agree that it is more correct to use `window` for this in Draft, not `global`. Didn't occur to me that it might make a difference in nwjs and electron though – I was only thinking of jsdom.

**Comment by LiaoSept** [comment]
@zpao based-browserify build

**Comment by spicyj** [comment]
This is still something Draft could do better so I'd like to keep this issue open.

**Comment by LiaoSept** [comment]
@spicyj how are you going to deal with this problem?

**Comment by rmccue** [comment]
I was getting this when trying to build a component around Draft and building it with Webpack. The shim that Webpack was adding for `global` (`(function () { return this; }())`) was giving `undefined` for some reason (potentially `"use strict"`).

Long story short, I fixed it for my usage by adding the following to my Webpack config:
```
plugins: [
	new webpack.DefinePlugin({
		"global": "window",
	}),
]
```
